scripts/migrate.js

```diff
@@ -0,0 +1,60 @@
+import AWS from 'aws-sdk';
+import Redis from 'ioredis';
+import { info } from 'heroku-logger';
+
+import Link from '../src/Models/Link';
+import { Client as PostgreSQL } from 'pg';
+import { filesystem } from '../src/helpers';
+
+// Configure storage connections for AWS, Redis, and RDS:
+AWS.config.update({ region: 'us-east-1' });
+
+const redis = new Redis(process.env.REDIS_URL);
+
+const sql = new PostgreSQL({
+  host: process.env.DB_HOST,
+  user: process.env.DB_USER,
+  password: process.env.DB_PASSWORD,
+  database: 'bertly_clicks',
+  port: 5432,
+});
+
+(async () => {
+  await sql.connect();
+
+  // Iterate over our existing links and migrate them to Bertly 2.0's storage:
+  const stream = redis.scanStream({ match: 'bertly:key:*', count: 100 });
```

Comment:
Is this `count: 100` what lets you get the `chunk` in the next line?

Reply:
Yep, the `SCAN` Redis command is basically the same idea as Laravel's `chunk()`. It grabs records in a batch of N, and then has a "next page" pointer that you can follow to request more after that. So this is saying to find `bertly:key:*` records in batches of 100.

It returns a [Node.js Stream](https://nodejs.org/api/stream.html) which we can then "iterate" over with `for await`.

Reply:
Woohoo thanks okay I'm with you on this bit now!